### PR TITLE
optimize staticpod controller for upgrade info

### PR DESCRIPTION
**Pull request by JameKeal**
/kind enhancement

#### What this PR does / why we need it:
There are some double counting in the existing code on upgrde info, and the calculation part is scattered and the parameter assignment is not uniform.
image
image

So this pr is used to optimized the upgrade info, and keep the computation in one place, rather than spreading it out.

Meanwhile, peel off the relevant calculation from staticpod_controller to upgrade_info, make the responsibilities clearer.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/assign @rambohe-ch 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@JameKeal please fix the unit tests error

**Comment by rambohe-ch** [comment]
@xavier-hou PTAL

File: pkg/controller/staticpod/upgradeinfo/upgrade_info.go [hunk 1]
```diff
@@ -128,64 +179,52 @@ func ReadyUpgradeWaitingNodes(infos map[string]*UpgradeInfo) []string {
...
-	for _, info := range infos {
-		if info.StaticPod != nil {
-			if podutils.IsPodReady(info.StaticPod) {
-				readyNumber++
-			}
+		// sync worker pods info
+		if info.WorkerPodDeleteNeeded {
+			deletePods = append(deletePods, info.WorkerPod)
+		}
+		if info.WorkerPodStatusPhase != corev1.PodFailed && info.WorkerPodStatusPhase != corev1.PodSucceeded {
```
Comment (xavier-hou) [inline pkg/controller/staticpod/upgradeinfo/upgrade_info.go:225]: Do we need to consider the existence of workerpod?
In this situation,`info.WorkerPodStatusPhase` may be "", and the result of `allSucceeded` is false

File: pkg/controller/staticpod/upgradeinfo/upgrade_info.go [hunk 2]
```diff
@@ -128,64 +179,52 @@ func ReadyUpgradeWaitingNodes(infos map[string]*UpgradeInfo) []string {
 func ReadyNodes(infos map[string]*UpgradeInfo) []string {
```
Comment (xavier-hou) [inline pkg/controller/staticpod/upgradeinfo/upgrade_info.go:179]: It seems that this function is not used. How about removing it?

File: pkg/controller/staticpod/upgradeinfo/upgrade_info.go (hunk 1 above)
Comment (JameKeal) [reply pkg/controller/staticpod/upgradeinfo/upgrade_info.go:225]: yes, you are right.

**Comment by xavier-hou** [comment]
@rambohe-ch LGTM  @JameKeal Plz fix the UT

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**